hazelcast/src/test/java/com/hazelcast/scheduledexecutor/ScheduledExecutorSplitBrainTest.java

```diff
@@ -110,6 +117,13 @@ protected void onAfterSplitBrainCreated(HazelcastInstance[] firstBrain, Hazelcas
             instance.getLifecycleService().addLifecycleListener(mergeLifecycleListener);
         }
 
+        // Wait for a few seconds, to allow the event system to finish delivering partition-lost events.
+        // This minimizes the chances of a race condition between handling the event and scheduling a new task.
+        // The IScheduledExecutor allows for tasks to be aware of lost partitions (marking the tasks as stale), hence, if
+        // we schedule a task before handling the event, then the task becomes stale (due to the event).
+        // Similar too: https://github.com/hazelcast/hazelcast/issues/12424
+        sleepSeconds(5);
```

Comment:
Fix for the race condition

Reply:
can't we eventually assert on proxies here? we can check if they set the flag or not